src/app/store/machine/utils/hooks.ts

```diff
@@ -32,6 +32,51 @@ import vlanSelectors from "app/store/vlan/selectors";
 import { isId } from "app/utils";
 import type { FilterSelected } from "app/utils/search/filter-items";
 
+export const useFetchMachineCount = (
+  filters?: FetchFilters
+): {
+  machineCount: number;
+  machineCountLoading: boolean;
+  machineCountLoaded: boolean;
+} => {
+  const [callId, setCallId] = useState<string | null>(null);
+  const previousCallId = usePrevious(callId);
+  const previousFilters = usePrevious(filters);
+  const dispatch = useDispatch();
+  const machineCount = useSelector((state: RootState) =>
+    machineSelectors.count(state, callId)
+  );
+  const machineCountLoading = useSelector((state: RootState) =>
+    machineSelectors.countLoading(state, callId)
+  );
+  const machineCountLoaded = useSelector((state: RootState) =>
+    machineSelectors.countLoaded(state, callId)
+  );
+
+  useEffect(() => {
+    if (!callId) {
+      setCallId(nanoid());
+    }
+  }, [callId, dispatch]);
+
+  useEffect(() => {
+    if (
+      (callId && callId !== previousCallId) ||
+      (callId &&
+        (filters || previousFilters) &&
+        !fastDeepEqual(filters, previousFilters))
+    ) {
+      dispatch(machineActions.count(callId, filters));
+    }
+  }, [dispatch, callId, previousCallId, filters, previousFilters]);
```

Comment:
This reuses the same callId which could result in race conditions if the calls are made in quick succession (similar to what was happening [in my cleanup PR](https://github.com/canonical/maas-ui/pull/4319#issuecomment-1210013605)). It also needs to clean up the request when the callId changes (though doesn't need to unsubscribe like the list/get requests).

Maybe something like this (copied from https://github.com/canonical/maas-ui/pull/4319):

```suggestion
  useEffect(() => {
    // undefined, null and {} are all equivalent i.e. no filters so compare the
    // current and previous filters using an empty object if the filters are falsy.
    if (!fastDeepEqual(filters || {}, previousFilters || {}) || !callId) {
      setCallId(nanoid());
    }
  }, [callId, dispatch, filters, previousFilters]);
  
  useEffect(() => {
    return () => {
      if (callId) {
        dispatch(machineActions.removeRequest(callId));
      }
    };
  }, [callId, dispatch]);

  useEffect(() => {
    if (callId && callId !== previousCallId) {
      dispatch(
        machineActions.count(callId, filters)
      );
    }
  }, [dispatch, filters, callId, previousCallId]);
```